test/machineslib.py

```diff
@@ -42,21 +42,18 @@ def waitPageInit(self):
         with self.browser.wait_timeout(30):
             self.browser.wait_in_text("body", "Virtual machines")
 
-    def performAction(self, vmName, action, checkExpectedState=True, connectionName="system"):
+    def performAction(self, vmName, action, checkExpectedState=True, connectionName="system", logPath=None):
         b = self.browser
         m = self.machine
 
-        def getStartTime():
-            return m.execute(f"grep 'starting up' /var/log/libvirt/qemu/{vmName}.log | tail -1")
-
-        initialTime = getStartTime()
         b.click(f"#vm-{vmName}-{connectionName}-action-kebab button")
         b.wait_visible(f"#vm-{vmName}-{connectionName}-action-kebab > .pf-v5-c-dropdown__menu")
         b.click(f"#vm-{vmName}-{connectionName}-{action} a")
-        if (action == "reboot" or "forceReboot") and connectionName == "system":
+        if (action == "reboot" or "forceReboot") and connectionName == "system" and logPath:
             # Check VM doesn't get rebooted after opening dialog
             # https://bugzilla.redhat.com/show_bug.cgi?id=2221144
-            self.assertEqual(initialTime, getStartTime())
+            # Need to clear the log before invoking
+            wait(lambda: "Linux version" not in m.execute(f"cat {logPath}"))
```

Comment:
ok I see you point now. My worry is that this is promt to a race condition. The "Linux version" will re-appear eventually in the log. wait retries very often, but it's really a matter of correct timing for this to work.

Reply:
right, and this is especially true as "Linux version" is pretty much the first thing that the kernel writes at boot. It would be better with checking for the login prompt. Unfortunately this is hard to avoid in general -- waiting until the log file shrinks in size will have the same problem.